streaming/base/dataset.py

```diff
@@ -666,23 +666,29 @@ def __iter__(self) -> Iterator[Dict[str, Any]]:
         for sample_id in self._each_sample(sample_ids):
             yield self[sample_id]
 
-    def state_dict(self, sample_in_epoch: int) -> Dict[str, Any]:
+    def state_dict(self, num_samples: int, from_beginning: bool) -> Dict[str, Any]:
         """Get a dict containing training state (called from non-worker process).
 
         This is called on rank zero.
 
         Args:
-            sample_in_epoch (int): The number of samples processed so far in the current epoch.
+            num_samples (int): The number of samples processed so far in the current epoch.
+            from_beginning (int): Whether we are counting from start of this epoch, or start of
+                just this (potentially resumed) training run this epoch.
 
         Returns:
             Dict[str, Any]: The state.
         """
         world = World()
         epoch = self.next_epoch - 1
         epoch, offset = self._resume(world, epoch)
+        if from_beginning:
```

Comment:
maybe add a comment here saying that `from_beginning = True` is used when using `StreamingDataloader` ??

Reply:
added note